gk/fib.c

```diff
@@ -1209,44 +1209,58 @@ init_gateway_fib_locked(struct ip_prefix *ip_prefix, enum gk_fib_action action,
 }
 
 static struct gk_fib *
-init_grantor_fib_locked(
-	struct ip_prefix *ip_prefix, struct ipaddr *gt_addr,
-	struct ipaddr *gw_addr, struct gk_config *gk_conf)
+init_grantor_fib_locked(struct ip_prefix *ip_prefix,
+	struct ipaddr *gt_addrs, struct ipaddr *gw_addrs,
+	unsigned int num_addrs, struct gk_config *gk_conf)
 {
+#define MAX_NUM_GRANTORS_PER_ENTRY \
+	((1 << (sizeof(gt_fib->u.grantor.set->num_entries) * 8)) - 1)
```

Comment:
Replace the `sizeof(...)` with with `RTE_SIZEOF_FIELD(struct gk_fib, u.grantor.set->num_entries)`, and move this define just before `init_grantor_fib_locked()`.